app/models/product.rb

```diff
@@ -0,0 +1,6 @@
+class Product < ApplicationRecord
+  validates :name, :description, :price, :category_id, presence: true
```

Comment:
I generally like to put each validation on its own line for readability. Also, in rails 5 `belongs_to` will validate the presence of the foreign key unless you specify `optional: true`

Reply:
Gotcha. I think I added that validation in there because of the error messages I was seeing in the tests. In this context without that validation, the error message is "must exist" rather than "can't be blank". I'll modify the test file to account for a category not existing thanks!